### ISSUE TITLE
[STM32] ADCAll and rtc.wakeup() not working on H743, probably related

### ISSUE THREAD

**Issue by rolandvs** [micropython/micropython#4323]
Using the latest version of the main branch and run the script on various boards yield into different results:

PYBV11 works
STM32F429DISC works
STM32L496DISC works
NUCLEO_H743 fails on ADCAll and rtc.wakeup()

_When using the script you have to press the user button after reset during 5 seconds in order to start the main()._

```
import pyb
import micropython
import sys

# run clock 
rtc=pyb.RTC()
rtc.init()

adc=pyb.ADCAll(12, 0x70000)

def gmv(tim):
        rtc.wakeup(tim)
        pyb.delay(10)
        pyb.LED(2).on()
        pyb.stop()
        rtc.wakeup(None)
    
def main():
    # test one
    print("   Platform:", sys.platform)
    print("RTC running: ", rtc.datetime())
    # test two
    print("ADC")
    print("Core TEMP", adc.read_core_temp())
    print("Core VBAT", adc.read_core_vbat())
    print("Core VREF", adc.read_core_vref())
    print("     VREF", adc.read_vref())
    # test three
    print("Here we test the wakeup function together with the pyb.sleep()")
    print("Program in endless loop, UART doesn't work anymore, LED3 blinks")

    while True:
        gmv(1000)
        pyb.LED(2).off()
        pyb.delay(500)

if __name__ == "__main__":

    print("Micropython")
    print("delay for 5 seconds")
    pyb.delay(5000)
    if pyb.Switch().value():
        pyb.delay(100)
        while pyb.Switch().value():
            pass
        main()
    else:
        pyb.LED(1).on()
        pyb.delay(100)
        pyb.LED(1).off()
    # exit of script
```

**Comment by Meekdai** [comment]
In the STM official reference manual, I found the temperature of the STM32H743 in the 18 channel of the ADC3. (VBAT is 17 & VREF is 19). Obviously its channel is different from STM32F4
[http://docs.micropython.org/en/latest/library/pyb.ADC.html?highlight=adcall#the-adcall-object](url)

**Comment by dpgeorge** [comment]
I can confirm the problems, and there are multiple issues: the H7 uses 16-bit ADC so the formulas for the calibration need to change, the sampling time for the temp sensor must be increased, and the RTC EXTI numbers need to be changed for the H7.  I have some working patches that will be pushed soon.

**Comment by dpgeorge** [comment]
The issues here should all be fixed by the following commits: 1b4031ed64bcc8cb207fe7b9527a59d527ada5ab (for RTC wakeup), 0555ada277c5cf84f4dd5dd2dd99b17f101dca78 (for reading of vref and vbat), 6cab8daee0c8ea5d409419781d806daa05c5babe (for reading of temp).
